### ISSUE TITLE
An error occurred during reasoning: Fresh entity policy: ALLOW Individual node set policy: BY_NAME timeout: 9223372036854775807.

### ISSUE THREAD

**Issue by Sebastian1992** [ontop/ontop#750]
### Description

I would like to extend a Microsoft SQL Server 2017 of an Automation Cell with Ontop for building a VKG. 
I am using Protege V.5.6.1 wiht the installed Ontop plug in. 
The database has many different tables. The devices or other objects are related to each other via primary and foreign key relations. 

But if I am using queries with JOINS and AS,  after starting the reasoner I always get the issue: **An error occurred during reasoning: Fresh entity policy: ALLOW Individual node set policy: BY_NAME timeout: 9223372036854775807.**

I am sorry, I am not so familiar with SQL and Ontop. Just starting in this field. 
Would Appreciate your help a lot.

### Steps to Reproduce

1. Created Ontology with Protege 
2. Connect to database successfully 
3. Run some firstsimple mappings 
4. Started to run mappings with JOINS 
5. Error occurs
**Expected behavior:** Reasoner in Protege will not fail. 

**Actual behavior:** An error occurred during reasoning: Fresh entity policy: ALLOW Individual node set policy: BY_NAME timeout: 9223372036854775807.

**Reproduces how often:** Anytime

### Attached material
Inside of the mapper the query is shown the result without any issue. 
<img width="510" alt="mapping result 2" src="https://github.com/ontop/ontop/assets/81219789/bcb6caa5-c1ff-42c9-8704-8103361086d1">

Used SQL query: 

SELECT
    D.[DeviceID],
    D.[SystemID],
    D.[DeviceType],
    D.[DeviceNumber],
    D.[DeviceName],
    P.[IPAddress]
FROM (
    SELECT [DeviceID], [SystemID], [DeviceType], [DeviceNumber], [DeviceName]
    FROM [Makino2].[dbo].[Devices]
    WHERE [DeviceType] = '2'
) AS D
JOIN (
    SELECT [DeviceID], [IPAddress]
    FROM (
        SELECT [DeviceID], [ParameterName], [ParameterValue]
        FROM [Makino2].[dbo].[DeviceParameters]
        WHERE [DeviceID] = '1' 
    ) AS SourceTable
    PIVOT (
        MAX([ParameterValue]) FOR [ParameterName] IN ([IPAddress])
    ) AS PivotTable
) AS P
ON D.[DeviceID] = P.[DeviceID];

### Versions

Protege V.5.6.1
with the Plug-In: **it.unibz.inf.ontop.protege-4.2.2.jar**
Microsoft SQL Server 2017

### Additional Information

Any additional information, configuration or data that might be necessary to reproduce the issue.

**Comment by kontchakov** [comment]
Hi Sebastian

Ontop 4.2.2 is nearly a year old. Could you please try version 5.1?

Also, if you could avoid using PIVOT in your SQL query, that would allow Ontop to parse your mapping query and deal with it more efficienltly.

Thanks
Roman

**Comment by apano-on** [comment]
You should also try specifying the RDF datatype, in your case string, in the target part of your mapping i.e.:
```
:MiloFMS/RailMountedVehicle/{DeviceID} a :RailMountedVehicle ; :systemName {DeviceName}^^xsd:string .
```
Your problem and cause seem similar to issue #469. The more informative error message should appear directly in version 5.1.0.

**Comment by Sebastian1992** [comment]
@kontchakov @apano-on  Thank you very much for this super fast response. 
Specifying the RDF datatype solved the issue also in my current version (4.2.2) of ontop.

But I will also move to the newer version of Ontop like suggested. It is good to hear that the error message is then more comprehensive. 

Sorry for bothering you with this "simple" issue. 

Again thanks a lot.

**Comment by Sebastian1992** [comment]
After great support it is working.
